app/src/main/java/protect/card_locker/MainActivity.java

```diff
@@ -532,18 +530,36 @@ public boolean onOptionsItemSelected(MenuItem inputItem)
                 }
             }
 
-            builder.setSingleChoiceItems(R.array.sort_types_array, currentIndex.get(), (dialog, which) -> currentIndex.set(which));
-            builder.setPositiveButton(R.string.sort, (dialog, which) -> {
-                setSort(loyaltyCardOrders.get(currentIndex.get()), DBHelper.LoyaltyCardOrderDirection.Ascending);
+            AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
+            builder.setTitle(R.string.sort_by);
 
-                dialog.dismiss();
+            final View customLayout = getLayoutInflater().inflate(R.layout.sorting_option, null);
+            builder.setView(customLayout);
+
+            CheckBox ch = (CheckBox) customLayout.findViewById(R.id.checkBox_reverse);
+            ch.setChecked(getSharedPreferences(getString(R.string.sharedpreference_reverse_state), Context.MODE_PRIVATE).getBoolean("ch", true));
```

Comment:
Please use the already existing sharedpreference_sort_direction

Reply:
Should I change it to sharedpreference_checkbox_state instead to avoid the confusion?

Reply:
No, you should read sharedpreference_sort_direction and base the checked state on that